tequilapi/endpoints/identities_test.go

```diff
@@ -1,20 +1,117 @@
 package endpoints
 
 import (
+	"net/http"
 	"net/http/httptest"
 	"testing"
 
+	"bytes"
 	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/server"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestListIdentities(t *testing.T) {
+const identityUrl = "/irrelevant"
 
-	req := httptest.NewRequest("GET", "/irrelevant", nil)
+var (
+	mockIdm    = identity.NewIdentityManagerFake()
+	mystClient = server.NewClientFake()
+)
+
+func TestRegisterExistingIdentityRequest(t *testing.T) {
+	req, err := http.NewRequest(
+		http.MethodPut,
+		identityUrl,
+		bytes.NewBufferString(
+			`{
+				"registered": false
+			}`))
+
+	assert.Nil(t, err)
+	resp := httptest.NewRecorder()
+
+	handlerFunc := NewIdentitiesEndpoint(mockIdm, mystClient).Register
+	handlerFunc(resp, req, nil)
+
+	assert.Equal(t, http.StatusNotImplemented, resp.Code)
+	assert.JSONEq(
+		t,
+		`{
+            "message": "Unregister not supported"
+        }`, resp.Body.String())
+}
+
+func TestRegisterIdentitySuccess(t *testing.T) {
+	req, err := http.NewRequest(
+		http.MethodPut,
+		identityUrl,
+		bytes.NewBufferString(
+			`{
+				"registered": true
+			}`))
+
+	assert.Nil(t, err)
 	resp := httptest.NewRecorder()
 
 	mockIdm := identity.NewIdentityManagerFake()
-	handlerFunc := NewIdentitiesEndpoint(mockIdm).List
+	handlerFunc := NewIdentitiesEndpoint(mockIdm, mystClient).Register
+	handlerFunc(resp, req, nil)
+
+	assert.Equal(t, http.StatusAccepted, resp.Code)
+}
+
+func TestCreateNewIdentityNoPassword(t *testing.T) {
+	req, err := http.NewRequest(
+		http.MethodPost,
+		"/identities",
+		bytes.NewBufferString(
+			`{
+                "password": ""
+            }`))
+
+	assert.Nil(t, err)
+
+	resp := httptest.NewRecorder()
+	handlerFunc := NewIdentitiesEndpoint(mockIdm, mystClient).Create
+	handlerFunc(resp, req, nil)
+
+	assert.JSONEq(
+		t,
+		`{
+            "message" : "validation_error",
+            "errors" : {
+                "password" : [ {"code" : "required" , "message" : "Field is required" } ]
+            }
+        }`, resp.Body.String())
+}
+
+func TestCreateNewIdentity(t *testing.T) {
+	req, err := http.NewRequest(
+		http.MethodPost,
+		"/identities",
+		bytes.NewBufferString(
+			`{
+				"password": "mypass"
+			}`))
+	assert.Nil(t, err)
+
+	resp := httptest.NewRecorder()
+
+	handlerFunc := NewIdentitiesEndpoint(mockIdm, mystClient).Create
+	handlerFunc(resp, req, nil)
+
+	assert.JSONEq(
+		t,
+		`{
```

Comment:
Align JSON formatting style  with project style see `openvpn/service_discovery/dto/service_definition_test.go`

Reply:
no such file. I don't understand how you want to format this.